tests/ftqc/test_decomposition.py

```diff
@@ -192,3 +194,235 @@ def rot_mbqc(rot_xzx_gate):
+    def test_tape_wires_if_no_mp_wires(self, mp):
+
+        tape = qml.tape.QuantumScript([qml.H(1), qml.X(0), qml.Y(2)], measurements=[mp], shots=1000)
+
+        (transformed_tape,), _ = convert_to_mbqc_formalism(tape)
+
+        if mp.wires:
+            assert len(transformed_tape.measurements[0].wires) == len(mp.wires)
+        else:
+            assert len(transformed_tape.measurements[0].wires) == len(tape.wires)
+
+    def test_conversion_of_multi_wire_circuit(self):
+
+        dev = qml.device("lightning.qubit")
+
+        theta = 2.5
+        with qml.queuing.AnnotatedQueue() as q:
+            RotXZX(theta, 0, theta / 2, 0)
+            RotXZX(theta / 2, 0, theta / 4, 1)
+            qml.RZ(theta / 3, 0)
+            qml.X(0)
+            qml.H(1)
+            qml.S(1)
+            qml.Y(1)
+            qml.CNOT([0, 1])
+
+        base_tape = qml.tape.QuantumScript.from_queue(q)
+        tape = base_tape.copy(measurements=[qml.sample(wires=[0, 1])])
+        reference_tape = base_tape.copy(
+            measurements=[
+                qml.expval(qml.X(0)),
+                qml.expval(qml.X(1)),
+                qml.expval(qml.Y(0)),
+                qml.expval(qml.Y(1)),
+                qml.expval(qml.Z(0)),
+                qml.expval(qml.Z(1)),
+            ]
+        )
+
+        # after transform, only state prep and MCMs are present on the tape
+        (transformed_tape,), _ = convert_to_mbqc_formalism(tape)
+        expected_gates = (GraphStatePrep, qml.CZ, qml.measurements.MidMeasureMP, qml.X, qml.Y)
+        for op in transformed_tape.operations:
+            if isinstance(op, qml.ops.Conditional):
+                assert isinstance(op.base, (qml.X, qml.Z, qml.measurements.MidMeasureMP))
+            else:
+                assert isinstance(op, expected_gates)
+
+        # diagonalize final measurements, convert to the mbqc formalism, and execute
+        res = []
+        for obs in (qml.X, qml.Y, qml.Z):
+            ops = base_tape.operations + obs(0).diagonalizing_gates() + obs(1).diagonalizing_gates()
+            tape = base_tape.copy(
+                operations=ops, measurements=[qml.sample(wires=[0, 1])], shots=500
+            )
+            (mbqc_tape,), _ = convert_to_mbqc_formalism(tape)
+            (diagonalized_tape,), _ = diagonalize_mcms(mbqc_tape)
+
+            samples = qml.execute([diagonalized_tape], dev)[0]
+            for wire in (0, 1):
+                mp = qml.expval(obs(wire))
+                res.append(mp.process_samples(samples, wire_order=[0, 1]))
+
+        reference_result = qml.execute([reference_tape], dev)[0]
+
+        # analytic results, to 2 s.f., are (-0.40, 0.95, -0.37, -0.25, 0.82, 6.2e-17)
+        # an atol of 0.1 is not ideal for comparing to these results, but it's enough
+        # to catch changes that modify the results, and we have to choose here between
+        # very slow, or fairly noisy
+        assert np.allclose(res, reference_result, atol=0.1)
```

Comment:
Do we have an idea how often we will get a pure stochastic failure here?

Reply:
I occasionally saw a failure a few times while I was testing locally, but it usually passes. It's such a slow test, I don't really want to go over 500 shots. I'll pass a seed to the device.